swarm/src/lib.rs

```diff
@@ -282,6 +285,7 @@ where
     /// similar mechanisms.
     external_addrs: Addresses,
 
+    // TODO: I think this should move into the connection manager behaviour as well.
```

Comment:
Into _the_ behaviour or into _a_ behaviour. I was thinking the same thing when I thought about this topic. Keeping a record of banned peers can likely be a `NetworkBehaviour` itself IMO.